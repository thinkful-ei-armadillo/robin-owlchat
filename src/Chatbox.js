import React from 'react';

export default function Chatbox(props){
 const messageType = props.type === 'message' ? 'view': 'hidden';
 const typeOtherThanMessage = props.type === 'message' ? 'hidden': 'view';
 const type = props.type === 'message' ? '' : props.type;
 
 return (
 <div className="chat">
 <div className="messages">
     <span className="chat-time-indicator" data-type-timestamp={props.timestamp}></span>
     <div className={messageType}>
       <p>{props.message}</p>
     </div>
     <div className={typeOtherThanMessage}>
         <strong>{props.name}</strong>
         "gave a {type}"
         <span name={props.type} className={props.type}></span>
     </div>
 </div>
</div>
)
}




