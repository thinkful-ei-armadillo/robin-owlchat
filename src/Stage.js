import React from 'react';

export default function Stage (props){
    const session = props.onStage ? 'stage view' : 'stage hidden';
return (
    <div className={session}>
        <div className="contact-on-stage" data-tooltip={props.name}>
            <div className="contact-on-stage-video-container">
                <img src={props.avatar} alt={props.name} className="user-avatar contact-on-stage-avatar">
                </img>
                <div className="contact-on-stage-details">
                    <span className="contact-on-stage-details-text">
                    {props.name}
                    </span>
                    <button className="contact-on-stage-details-button"></button>
                </div>
            </div>
        </div>
    </div>
)
}