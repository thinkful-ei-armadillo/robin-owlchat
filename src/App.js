import React, { Component } from 'react';
import './App.css';
import Stage from './Stage';
import Chatbox from './Chatbox';

class App extends Component {
  
  render() {
    const { participants, chatEvents } = this.props;
    return (
    <main>
      <div className="chatroom">
        {chatEvents.map(person => (
          <Chatbox 
          key = {person.key}
          type={person.type} 
          message={person.message} 
          timestamp={person.timestamp} 
          messagetype ={person.messagetype}
          />
        ))} 
      </div>
      <div className="stage">
         {participants.map(person => (
           <Stage 
           name = {person.id}
           avatar = {person.avatar}
           onStage ={person.onStage}
           />))}
      </div>
    </main>
    );
  }
}

export default App;
