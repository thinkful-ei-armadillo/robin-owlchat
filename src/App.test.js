import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import Chatbox from './Chatbox';
import Stage from './Stage';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const participants= [{  id: 5,
    name: 'Ashla Attwool',
    avatar:
        'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true

  }] ;
  const chatEvents=[{
    participantId: 3,
        type: 'message',
        message: 'Hello world',
        time: 1548852646559,
        timestamp: 1548852484247
  }];
  ReactDOM.render(<App participants = {participants} chatEvents={chatEvents} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const participants= [{  id: 5,
    name: 'Ashla Attwool',
    avatar:
        'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true

  }] ;
  const chatEvents=[{
    participantId: 3,
        type: 'message',
        message: 'Hello world',
        time: 1548852646559,
        timestamp: 1548852484247
  }];
  const tree = renderer
    .create(<app participants = {participants} chatEvents={chatEvents}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });