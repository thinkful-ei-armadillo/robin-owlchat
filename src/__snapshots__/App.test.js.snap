// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<app
  chatEvents={
    Array [
      Object {
        "message": "Hello world",
        "participantId": 3,
        "time": 1548852646559,
        "timestamp": 1548852484247,
        "type": "message",
      },
    ]
  }
  participants={
    Array [
      Object {
        "avatar": "https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1",
        "id": 5,
        "inSession": true,
        "name": "Ashla Attwool",
        "onStage": true,
      },
    ]
  }
/>
`;
